web/vite.config.ts

```diff
@@ -10,6 +10,20 @@ export default defineConfig(() => ({
     svelte(),
     nodeResolve({
       extensions: ['.js', '.ts']
-    })
+    }),
+    {
+      enforce: 'pre' as const,
+      // Deploy section.
+      name: 'html-transform',
+      // Changing the favicon depending on the environment mode
+      transformIndexHtml: (html: string) => {
```

Comment:
We don’t need `: string` type here because `defineConfig()` already defined types for its argument.

So TS will know the types of `transformIndexHtml`.

```suggestion
      transformIndexHtml(html) {
```

And by removing `: string` you can now remove `as const` too. The issue with types was here because we re-define `transformIndexHtml` types.